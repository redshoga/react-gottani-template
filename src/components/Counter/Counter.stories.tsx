import React from "react";
import { Counter } from "./Counter";

export default {
  title: "Counter",
  component: Counter
};

const mockProps = {
  counter: 123,
  increment: () => {},
  decrement: () => {}
};

export const Default = () => <Counter {...mockProps} />;
