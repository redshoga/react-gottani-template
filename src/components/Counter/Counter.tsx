import React from "react";

export type Props = {
  counter: number;
  increment: () => void;
  decrement: () => void;
};

export enum TestAttr {
  Counter = "Counter",
  IncrementButton = "IncrementButton",
  DecrementButton = "DecrementButton"
}

export class Counter extends React.Component<Props> {
  render() {
    const { increment, decrement, counter } = this.props;
    return (
      <>
        <div>
          <button data-test={TestAttr.IncrementButton} onClick={increment}>
            increment
          </button>
        </div>
        <div>
          <button data-test={TestAttr.DecrementButton} onClick={decrement}>
            decrement
          </button>
        </div>
        count: <span data-test={TestAttr.Counter}>{counter}</span>
      </>
    );
  }
}
