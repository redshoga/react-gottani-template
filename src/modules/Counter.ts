import { Action } from "redux";

// action types
const INCREMENT = "COUNTER_INCREMENT";
const DECREMENT = "COUNTER_DECREMENT";

interface IncrementAction extends Action {
  type: typeof INCREMENT;
}

interface DecrementAction extends Action {
  type: typeof DECREMENT;
}

export type ActionTypes = IncrementAction | DecrementAction;

// action creator
const increment: () => ActionTypes = () => {
  return { type: INCREMENT };
};

const decrement: () => ActionTypes = () => {
  return { type: DECREMENT };
};

export const actionCreator = {
  increment,
  decrement
};

// state
export interface State {
  count: number;
}
export const initialState: State = {
  count: 123
};

// reducer
export function reducer(state = initialState, action: ActionTypes): State {
  switch (action.type) {
    case INCREMENT:
      return {
        ...state,
        count: state.count + 1
      };
    case DECREMENT:
      return {
        ...state,
        count: state.count - 1
      };
    default:
      return state;
  }
}
