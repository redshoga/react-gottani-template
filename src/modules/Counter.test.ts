import reducerTester from "reducer-tester";
import { reducer, initialState, actionCreator } from "./Counter";

describe("Counter", () => {
  reducerTester({
    reducer,
    state: initialState,
    tests: [actionCreator.decrement(), actionCreator.increment()]
  });
});
