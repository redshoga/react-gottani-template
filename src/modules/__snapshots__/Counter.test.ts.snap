// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Counter/DECREMENT 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
-   "count": 123,
+   "count": 122,
  }
`;

exports[`Counter Counter/INCREMENT 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
-   "count": 123,
+   "count": 124,
  }
`;

exports[`Counter handle initial state 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;
